### PR TITLE
Bug 1068987 - aria-hide main gallery views when dialogs appear.

### PR DESCRIPTION
**Pull request by eeejay**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=961114dae254)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=ea9efbbfed7e)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=24b6d0f7cef6)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=cae53a754f5d)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=23168f49b80d)

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=044adf24afec)

File: apps/gallery/style/gallery.css [hunk 1]
```diff
@@ -41,6 +41,10 @@ section {
   display: none;
 }
 
+body.showing-dialog > section {
```
Comment (davidflanagan) [inline apps/gallery/style/gallery.css]: To be honest, I don't actually understand why this is necessary. Is the screen reader not able to tell that these elements are obscured by the opaque dialog? Is that a bug in the screen reader?  Is it normal to have to do something like this for accessible dialogs, or is there something weird about the way that we do dialogs in this app that is forcing us to do it this way?

If you could explain these issues in a comment here in the gallery.css file, that would be really nice.

File: apps/gallery/style/gallery_tablet.css [hunk 2]
```diff
@@ -339,7 +339,7 @@ body.fullscreenView #fullscreen-back-button-large {
  */
 #overlay,
 #confirm-dialog {
-  background: rgba(51, 51, 51, 0.6);
+  background: rgb(51, 51, 51);
```
Comment (davidflanagan) [inline apps/gallery/style/gallery_tablet.css]: I think this is more concisely expressed as #333, if I did the conversion correctly.

File: tests/python/gaia-ui-tests/gaiatest/apps/gallery/app.py [hunk 3]
```diff
@@ -21,6 +21,7 @@ class Gallery(Base):
     name = 'Gallery'
 
     _gallery_items_locator = (By.CSS_SELECTOR, 'div.thumbnail')
+    _overlay_locator = (By.ID, 'overlay')
```
Comment (davidflanagan) [inline tests/python/gaia-ui-tests/gaiatest/apps/gallery/app.py]: These python test changes seem good, but also seem unrelated to this patch. Did you mean for them to be here, or is this just a merge conflict sort of thing?

File: shared/js/dialogs.js [hunk 4]
```diff
@@ -46,10 +46,17 @@ var Dialogs = {
         confirmButton.classList.remove('danger');
       }
 
+      if (options.bodyClass) {
```
Comment (davidflanagan) [inline shared/js/dialogs.js]: I'm always nervous about changes to shared code, but this one is clearly opt-in, so it seems safe. I would like a comment before this line:

// If the caller specified a bodyClass, add that to the classList of the body element.
// This can be used or accessibility, e.g., to hide the main document while a dialog is shown

File: apps/gallery/style/gallery.css (hunk 1 above)
Comment (eeejay) [inline apps/gallery/style/gallery.css]: I'll add a comment  about that in the CSS. The screen reader does not distinguished between obscured and un-obscured content. First, because it is really hard to do, you need to take into account the relative opacity of each obscuring layout frame, and false positives would be very bad. Also, historically developers would hide text intended for screen readers in different fashions, like putting it in a negative offset, using font-size:0 or obscuring it. The issue of modal dialogs is not unique to this module, and is the most painful part of making gaia accessible. We will need to do this until the <dialog> element is implemented in gecko.

File: tests/python/gaia-ui-tests/gaiatest/apps/gallery/app.py (hunk 3 above)
Comment (eeejay) [inline tests/python/gaia-ui-tests/gaiatest/apps/gallery/app.py]: Yes. now that we actually hide the main sections when the overlay is displayed, you can't depend on the thumbnail view when there are no photos. You need to wait for the overlay. So I updated the tests since my changes "break" them.

Comment (eeejay) [inline tests/python/gaia-ui-tests/gaiatest/apps/gallery/app.py]: Yes. now that we actually hide the main sections when the overlay is displayed, you can't depend on the thumbnail view when there are no photos. You need to wait for the overlay. So I updated the tests since my changes "break" them.

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=39c37258f5fa)

File: shared/js/dialogs.js [hunk 5]
```diff
@@ -46,10 +46,20 @@ var Dialogs = {
         confirmButton.classList.remove('danger');
       }
 
+      // If the caller specified a bodyClass, add that to the classList of the
+      // body element. This can be used or accessibility, e.g., to hide the main
+      // document while a dialog is shown
```
Comment (punamdahiya) [inline shared/js/dialogs.js]: nit: typo should be used for accessibility

**Comment by try-server-hook** [comment]
[eeejay Eitan Isaacson (eeejay)](https://github.com/eeejay) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=46f0db063791)
